lib/manageiq/providers/ibm_cloud/cloud_tools/vpc_sdk/instance.rb

```diff
@@ -72,6 +72,19 @@ def delete
               @parent.request(:delete_instance, :id => id)
             end
 
+            def resize(instance, new_flavor_name)
+              @parent.request(:update_instance,
+                              :id             => instance[:id],
+                              :instance_patch => {
+                                :name    => instance[:name],
+                                :profile => {:name => new_flavor_name}
+                              }
+                             )
+            rescue IBMCloudSdkCore::ApiException => e
+              Notification.create(:type => :vm_resize_error, :options => {:message => e.to_s})
+              raise MiqException::MiqProvisionError, e.to_s
```

Comment:
I think this exception handler would be better to live up [[here]](https://github.com/ManageIQ/manageiq-providers-ibm_cloud/pull/300/files#diff-5af23959f8142218f0266befea8e232991173bc7a177a5b6a24154a782cc2c2eR117) so this stays more of a pure IBM Cloud interface and keep the MIQ bits local to `app/models`